app/main.py

```diff
@@ -2,24 +2,89 @@
 
 
 class IntegerRange:
-    pass
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.private_name = "_" + name
+
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(
+            self,
+            instance: object,
+            owner: object
+    ) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.private_name, value)
+        else:
+            raise ValueError
```

Comment:
Looks like used redundant else. You can try to use `if not (...)` and raise error, after that if place set attr, so if error not raised you can set it!